### PR TITLE
prepare assert-ids script

### PR DESCRIPTION
**Pull request by podborski**
update script to prepare assert-id's for AV1-ISOBMFF.

### CODE REVIEW

**Comment by podborski** [comment]
```shell
python aomrules.py --spec av1isobmff --dump
AOM specification: 'av1isobmff'

{
  "It SHALL conform to the normative requirements of [ISOBMFF]",
  "assert-3d78af2f",
                [](Box const & root, IReport * out)
  {
  }
},
{
  "It SHALL have the av01 brand among the compatible brands array of the FileTypeBox",
  "assert-03258f22",
                [](Box const & root, IReport * out)
  {
  }
},
{
  "It SHALL contain at least one track using an AV1SampleEntry, possibly transformed by encryption as specified in § 4 Common Encryption",
  "assert-bd1c6212",
                [](Box const & root, IReport * out)
  {
  }
},
{
  "It SHOULD indicate a structural ISOBMFF brand among the compatible brands array of the FileTypeBox, such as iso6",
  "assert-5e63f779",
                [](Box const & root, IReport * out)
  {
  }
},
{
  "It MAY indicate CMAF brands as specified in § 3 CMAF AV1 track format",
  "assert-c652c075",
                [](Box const & root, IReport * out)
  {
  }
},
{
  "It MAY indicate other brands not specified in this document provided that the associated requirements do not conflict with those given in this specification",
  "assert-384c2374",
                [](Box const & root, IReport * out)
  {
  }
},
{
  "Mandatory:         Yes for av01",
  "assert-485d25aa",
                [](Box const & root, IReport * out)
  {
  }
},
{
  "Quantity:          One or more for av01",
  "assert-e091fa3c",
                [](Box const & root, IReport * out)
  {
  }
},
{
  "The width and height fields of the VisualSampleEntry SHALL equal the values of max_frame_width_minus_1 + 1 and max_frame_height_minus_1 + 1 of the Sequence Header OBU applying to the samples associated with this sample entry.",
  "assert-4708372f",
                [](Box const & root, IReport * out)
  {
  }
},
{
  "The width and height in the TrackHeaderBox SHOULD be equal to MaxRenderWidth and MaxRenderHeight, respectively.",
  "assert-b8425a84",
                [](Box const & root, IReport * out)
  {
  }
},
{
  "Additionally, if MaxRenderWidth and MaxRenderHeight values do not equal respectively the max_frame_width_minus_1 + 1 and max_frame_height_minus_1 + 1 values of the Sequence Header OBU, a PixelAspectRatioBox box SHALL be present in the sample entry and set such that",
  "assert-54ae6192",
                [](Box const & root, IReport * out)
  {
  }
},
{
  "The value \012AOM Coding is RECOMMENDED",
  "assert-da9cc152",
                [](Box const & root, IReport * out)
  {
  }
},
{
  "The config field SHALL contain an AV1CodecConfigurationBox that applies to the samples associated with this sample entry.",
  "assert-8d3f8e0c",
                [](Box const & root, IReport * out)
  {
  }
},
{
  "Multiple instances of AV1SampleEntry may be required when the track contains samples requiring a AV1CodecConfigurationBox with different characteristics.",
  "assert-8d7a614c",
                [](Box const & root, IReport * out)
  {
  }
},
{
  "Optional boxes not specifically mentioned here can be present, in particular those indicated in the definition of the VisualSampleEntry in [ISOBMFF].",
  "assert-d9e74d75",
                [](Box const & root, IReport * out)
  {
  }
},
{
  "Mandatory: Yes for av1C",
  "assert-318390e9",
                [](Box const & root, IReport * out)
  {
  }
},
{
  "Quantity:  Exactly One for av1C",
  "assert-a249db05",
                [](Box const & root, IReport * out)
  {
  }
},
{
  "The AV1CodecConfigurationBox contains decoder configuration information that SHALL be valid for every sample that references the sample entry.",
  "assert-f875c695",
                [](Box const & root, IReport * out)
  {
  }
},
{
  "The marker field SHALL be set to 1.",
  "assert-52768b11",
                [](Box const & root, IReport * out)
  {
  }
},
{
  "The value SHALL be set to 1 for AV1CodecConfigurationRecord",
  "assert-49a325d3",
                [](Box const & root, IReport * out)
  {
  }
},
{
  "The seq_profile field indicates the AV1 profile and SHALL be equal to the seq_profile value from the Sequence Header OBU.",
  "assert-96a6c200",
                [](Box const & root, IReport * out)
  {
  }
},
{
  "The seq_level_idx_0 field indicates the value of seq_level_idx[0] found in the Sequence Header OBU and SHALL be equal to the value of seq_level_idx[0] in the Sequence Header OBU",
  "assert-7d134bb5",
                [](Box const & root, IReport * out)
  {
  }
},
{
  "The seq_tier_0 field indicates the value of seq_tier[0] found in the Sequence Header OBU and SHALL be equal to the value of seq_tier[0] in the Sequence Header OBU",
  "assert-3fe26d43",
                [](Box const & root, IReport * out)
  {
  }
},
{
  "When twelve_bit is not present in the Sequence Header OBU the AV1CodecConfigurationRecord twelve_bit value SHALL be 0.",
  "assert-71c21ca1",
                [](Box const & root, IReport * out)
  {
  }
},
{
  "When not specified in the Sequence Header OBU, and not defined by the conditions specified in the [AV1] color_config, the values of chroma_subsampling_x, chroma_subsampling_y, and chroma_sample_position SHALL be 0.",
  "assert-9d2dbc84",
                [](Box const & root, IReport * out)
  {
  }
},
{
  "the following procedure SHALL not return any error",
  "assert-00f2331b",
                [](Box const & root, IReport * out)
  {
  }
},
{
  "The configOBUs field contains zero or more OBUs",
  "assert-8890b1aa",
                [](Box const & root, IReport * out)
  {
  }
},
{
  "From any sync sample, an AV1 bitstream is formed by first outputting the OBUs contained in the AV1CodecConfigurationBox and then by outputing all OBUs in the samples themselves, in order, starting from the sync sample.",
  "assert-d046552a",
                [](Box const & root, IReport * out)
  {
  }
},
{
  "From any sample marked with the AV1ForwardKeyFrameSampleGroupEntry, an AV1 bitstream is formed by first outputting the OBUs contained in the AV1CodecConfigurationBox and then by outputing all OBUs in the sample itself, then by outputting all OBUs in the samples, in order, starting from the sample at the distance indicated by the sample group.",
  "assert-bb553a27",
                [](Box const & root, IReport * out)
  {
  }
},
{
  "the configOBUs field SHALL contain at most one Sequence Header OBU",
  "assert-755c9133",
                [](Box const & root, IReport * out)
  {
  }
},
{
  "it SHALL be the first OBU",
  "assert-b90b2cfc",
                [](Box const & root, IReport * out)
  {
  }
},
{
  "The flag obu_has_size_field SHALL be set to 1, indicating that the size of the OBU payload follows the header, and that it is coded using LEB128",
  "assert-cf9ef74c",
                [](Box const & root, IReport * out)
  {
  }
},
{
  "When a Sequence Header OBU is contained within the configOBUs of the AV1CodecConfigurationRecord, the values present in the Sequence Header OBU contained within configOBUs SHALL match the values of the AV1CodecConfigurationRecord.",
  "assert-745b4db3",
                [](Box const & root, IReport * out)
  {
  }
},
{
  "The timing_info_present_flag in the Sequence Header OBU (in the configOBUs field or in the associated samples) SHOULD be set to 0",
  "assert-551498bd",
                [](Box const & root, IReport * out)
  {
  }
},
{
  "If set to 1, the timing_info structure of the Sequence Header OBU, the frame_presentation_time and buffer_removal_time fields of the Frame Header OBUs, if present, SHALL be ignored for the purpose of timed processing of the ISOBMFF file.",
  "assert-9be6e647",
                [](Box const & root, IReport * out)
  {
  }
},
{
  "The sample entry SHOULD contain a colr box with a colour_type set to nclx",
  "assert-6056f4f8",
                [](Box const & root, IReport * out)
  {
  }
},
{
  "If any of colour_primaries, transfer_characteristics, and matrix_coefficients is specified with a value other than 2 in the Sequence Header OBU (in the configOBUs field or in the associated samples), then the value of that field in the colr box SHALL match the value in the Sequence Header OBU.",
  "assert-cb060b01",
                [](Box const & root, IReport * out)
  {
  }
},
{
  "If any of colour_primaries, transfer_characteristics, and matrix_coefficients is NOT specified or is specified with a value equal to 2 in the Sequence Header OBU, then the value of that field in the colr box MAY be different and the value in the colr box overrides the value in the bitstream, as specified in the ColourInformationBox definition in ISOBMFF.",
  "assert-deaa3a6a",
                [](Box const & root, IReport * out)
  {
  }
},
{
  "The value of full_range_flag in the colr box SHALL match the color_range flag in the Sequence Header OBU.",
  "assert-21d17459",
                [](Box const & root, IReport * out)
  {
  }
},
{
  "When configOBUs does not contain a Sequence Header OBU, a colr box with a colour_type set to nclx SHALL be present.",
  "assert-ae2ade7e",
                [](Box const & root, IReport * out)
  {
  }
},
{
  "The ContentLightLevelBox clli SHOULD be present",
  "assert-3e46dfaf",
                [](Box const & root, IReport * out)
  {
  }
},
{
  "Its values SHALL equal the values contained in the Metadata OBU of type METADATA_TYPE_HDR_CLL, if present (in the configOBUs or in the samples)",
  "assert-0707714d",
                [](Box const & root, IReport * out)
  {
  }
},
{
  "The MasteringDisplayColourVolumeBox mdcv SHOULD be present",
  "assert-2c71f659",
                [](Box const & root, IReport * out)
  {
  }
},
{
  "Its decoded values SHALL equal (except for precision issues, see the second note below) the decoded values contained in the Metadata OBU of type METADATA_TYPE_HDR_MDCV, if present (in the configOBUs or in the samples)",
  "assert-349bbb63",
                [](Box const & root, IReport * out)
  {
  }
},
{
  "Additional boxes may be provided at the end of the VisualSampleEntry as permitted by ISOBMFF, that may represent redundant or similar information to the one provided in some OBUs contained in the AV1CodecConfigurationBox",
  "assert-fa8b4082",
                [](Box const & root, IReport * out)
  {
  }
},
{
  "the sample data SHALL be a sequence of OBUs forming a Temporal Unit",
  "assert-9ba1392f",
                [](Box const & root, IReport * out)
  {
  }
},
{
  "each OBU SHALL follow the open_bitstream_unit Low Overhead Bitstream Format syntax as specified in [AV1]",
  "assert-f204884a",
                [](Box const & root, IReport * out)
  {
  }
},
{
  "Each OBU SHALL have the obu_has_size_field set to 1 except for the last OBU in the sample",
  "assert-f8d5b9b7",
                [](Box const & root, IReport * out)
  {
  }
},
{
  "obu_has_size_field MAY be set to 0, in which case it is assumed to fill the remainder of the sample",
  "assert-d277b0cf",
                [](Box const & root, IReport * out)
  {
  }
},
{
  "When extracting OBUs from an ISOBMFF file, and depending on the capabilities of the decoder processing these OBUs, ISOBMFF parsers MAY need to either: set the obu_has_size_field to 1 for some OBUs if not already set, add the size field in this case, and add Temporal Delimiter OBU; or use the length-delimited bitstream format as defined in Annex B of AV1.",
  "assert-00151a99",
                [](Box const & root, IReport * out)
  {
  }
},
{
  "If encryption is used, similar operations MAY have to be done before or after decryption depending on the demuxer/decryptor/decoder architecture.",
  "assert-e0aa5501",
                [](Box const & root, IReport * out)
  {
  }
},
{
  "OBU trailing bits SHOULD be limited to byte alignment and SHOULD not be used for padding",
  "assert-c2e52ab3",
                [](Box const & root, IReport * out)
  {
  }
},
{
  "OBUs of type OBU_TEMPORAL_DELIMITER, OBU_PADDING, or OBU_REDUNDANT_FRAME_HEADER SHOULD NOT be used",
  "assert-2487540d",
                [](Box const & root, IReport * out)
  {
  }
},
{
  "OBUs of type OBU_TILE_LIST SHALL NOT be used",
  "assert-c7a31be1",
                [](Box const & root, IReport * out)
  {
  }
},
{
  "If an AV1 Sample is signaled as a sync sample (in the SyncSampleBox or by setting sample_is_non_sync_sample to 0), it SHALL be a Random Access Point as defined in [AV1]",
  "assert-bee456d5",
                [](Box const & root, IReport * out)
  {
  }
},
{
  "Intra-only frames SHOULD be signaled using the sample_depends_on flag set to 2",
  "assert-0c895956",
                [](Box const & root, IReport * out)
  {
  }
},
{
  "Delayed Random Access Points SHOULD be signaled using sample groups and the AV1ForwardKeyFrameSampleGroupEntry",
  "assert-4f779503",
                [](Box const & root, IReport * out)
  {
  }
},
{
  "Switch Frames SHOULD be signaled using sample groups and the AV1SwitchFrameSampleGroupEntry",
  "assert-d10ee363",
                [](Box const & root, IReport * out)
  {
  }
},
{
  "if a file contains multiple tracks that are alternative representations of the same content, in particular using Switch Frames, those tracks SHOULD be marked as belonging to the same alternate group",
  "assert-ccbd7555",
                [](Box const & root, IReport * out)
  {
  }
},
{
  "SHOULD use a track selection box with an appropriate attribute (e.g. bitr)",
  "assert-7591fd9c",
                [](Box const & root, IReport * out)
  {
  }
},
{
  "In tracks using the AV1SampleEntry, the ctts box and composition offsets in movie fragments SHALL NOT be used",
  "assert-0f174d22",
                [](Box const & root, IReport * out)
  {
  }
},
{
  "the is_leading flag, if used, SHALL be set to 0 or 2",
  "assert-cb746c39",
                [](Box const & root, IReport * out)
  {
  }
},
{
  "When a temporal unit contains more than one frame, the sample corresponding to that temporal unit MAY be marked using the AV1MultiFrameSampleGroupEntry",
  "assert-227cd7b6",
                [](Box const & root, IReport * out)
  {
  }
},
{
  "Metadata OBUs may be carried in sample data",
  "assert-e8ac7f22",
                [](Box const & root, IReport * out)
  {
  }
},
{
  "the AV1MetadataSampleGroupEntry SHOULD be used",
  "assert-d41e5e3f",
                [](Box const & root, IReport * out)
  {
  }
},
{
  "If the metadata OBUs are static for the entire set of samples associated with a given sample description entry, they SHOULD also be in the OBU array in the sample description entry",
  "assert-f0ce5ae3",
                [](Box const & root, IReport * out)
  {
  }
},
{
  "Mandatory:  No for av1f",
  "assert-b19a22ce",
                [](Box const & root, IReport * out)
  {
  }
},
{
  "Quantity:   Zero or more for av1f",
  "assert-6de7aff7",
                [](Box const & root, IReport * out)
  {
  }
},
{
  "Mandatory:  No for av1m",
  "assert-5482ba80",
                [](Box const & root, IReport * out)
  {
  }
},
{
  "Quantity:   Zero or more for av1m",
  "assert-35d7f2e6",
                [](Box const & root, IReport * out)
  {
  }
},
{
  "Mandatory:  No for av1s",
  "assert-72c62575",
                [](Box const & root, IReport * out)
  {
  }
},
{
  "Quantity:   Zero or more for av1s",
  "assert-731712f5",
                [](Box const & root, IReport * out)
  {
  }
},
{
  "Mandatory:  No for av1M",
  "assert-b6627aa5",
                [](Box const & root, IReport * out)
  {
  }
},
{
  "Quantity:   Zero or more for av1M",
  "assert-fa0bc954",
                [](Box const & root, IReport * out)
  {
  }
},
{
  "In this version of the specification, metadata_specific_parameters is only defined when metadata_type is set to METADATA_TYPE_ITUT_T35 in which case its value SHALL be set to the first 24 bits of the metadata_itut_t35 structure.",
  "assert-7d13a03d",
                [](Box const & root, IReport * out)
  {
  }
},
{
  "For other types of metadata, its value SHOULD be set to 0. In all cases, when the grouping_type_parameter is used, readers processing sample groups SHALL use the entire value of grouping_type_parameter",
  "assert-973cddc9",
                [](Box const & root, IReport * out)
  {
  }
},
{
  "it SHALL use an AV1SampleEntry, possibly transformed by encryption as specified in § 4 Common Encryption",
  "assert-ffcf1af0",
                [](Box const & root, IReport * out)
  {
  }
},
{
  "it MAY use multiple sample entries, and in that case the following values SHALL not change in the track",
  "assert-aec6597a",
                [](Box const & root, IReport * out)
  {
  }
},
{
  "When protected, CMAF AV1 Tracks SHALL use the signaling defined in [CMAF], which in turn relies on [CENC], with the provisions specified in § 4 Common Encryption",
  "assert-7250f9ce",
                [](Box const & root, IReport * out)
  {
  }
},
{
  "CMAF AV1 Tracks and non-segmented AV1 files MAY be protected",
  "assert-6eb63a07",
                [](Box const & root, IReport * out)
  {
  }
},
{
  "If protected, they SHALL conform to [CENC]",
  "assert-89e67c25",
                [](Box const & root, IReport * out)
  {
  }
},
{
  "Files SHOULD be protected using the cbcs protection scheme",
  "assert-2a42f17e",
                [](Box const & root, IReport * out)
  {
  }
},
{
  "MAY be protected using the cenc protection scheme",
  "assert-105dfb43",
                [](Box const & root, IReport * out)
  {
  }
},
{
  "When the protected scheme cenc is used, samples SHALL be protected using subsample encryption",
  "assert-44221cba",
                [](Box const & root, IReport * out)
  {
  }
},
{
  "SHALL NOT use pattern encryption",
  "assert-aa4aefdb",
                [](Box const & root, IReport * out)
  {
  }
},
{
  "When the protected scheme cbcs is used, samples SHALL be protected using subsample encryption",
  "assert-0113b336",
                [](Box const & root, IReport * out)
  {
  }
},
{
  "SHALL use pattern encryption",
  "assert-95fbc893",
                [](Box const & root, IReport * out)
  {
  }
},
{
  "crypt_byte_block = 1 and skip_byte_block = 9",
  "assert-51e8a94d",
                [](Box const & root, IReport * out)
  {
  }
},
{
  "crypt_byte_block = 5 and skip_byte_block = 5",
  "assert-973b126b",
                [](Box const & root, IReport * out)
  {
  }
},
{
  "crypt_byte_block = 10 and skip_byte_block = 0",
  "assert-0fbb54b0",
                [](Box const & root, IReport * out)
  {
  }
},
{
  "Protected samples SHALL be exactly spanned by one or more contiguous subsamples",
  "assert-d84cddee",
                [](Box const & root, IReport * out)
  {
  }
},
{
  "An OBU MAY be spanned by one or more subsamples",
  "assert-38a31899",
                [](Box const & root, IReport * out)
  {
  }
},
{
  "A large unprotected OBU whose data size is larger than the maximum size of a single BytesOfClearData field MAY be spanned by multiple subsamples with zero size BytesOfProtectedData",
  "assert-02d6b0e9",
                [](Box const & root, IReport * out)
  {
  }
},
{
  "All OBU Headers and associated obu_size fields SHALL be unprotected",
  "assert-b8cabcad",
                [](Box const & root, IReport * out)
  {
  }
},
{
  "Temporal Delimiter OBUs, Sequence Header OBUs, and Frame Header OBUs (including within a Frame OBU), Redundant Frame Header OBUs and Padding OBUs SHALL be unprotected",
  "assert-4b9592fb",
                [](Box const & root, IReport * out)
  {
  }
},
{
  "Metadata OBUs SHALL NOT be protected",
  "assert-21c3c265",
                [](Box const & root, IReport * out)
  {
  }
},
{
  "Tile Group OBUs and Frame OBUs SHALL be protected",
  "assert-8750fa6f",
                [](Box const & root, IReport * out)
  {
  }
},
{
  "BytesOfProtectedData SHALL start on the first byte and end on the last byte of the decode_tile structure (including any trailing bits)",
  "assert-61f691fe",
                [](Box const & root, IReport * out)
  {
  }
},
{
  "A subsample SHALL be created for each tile, even if its size is less than 16 bytes",
  "assert-6547e5e3",
                [](Box const & root, IReport * out)
  {
  }
},
{
  "All other parts of Tile Group OBUs and Frame OBUs SHALL be unprotected for cbcs",
  "assert-68939155",
                [](Box const & root, IReport * out)
  {
  }
},
{
  "BytesOfProtectedData SHALL be a multiple of 16 bytes",
  "assert-da5e71ae",
                [](Box const & root, IReport * out)
  {
  }
},
{
  "BytesOfProtectedData SHALL end on the last byte of the decode_tile structure (including any trailing bits)",
  "assert-839432b2",
                [](Box const & root, IReport * out)
  {
  }
},
{
  "BytesOfProtectedData SHALL span all complete 16-byte blocks of the decode_tile structure (including any trailing bits)",
  "assert-b9de48fc",
                [](Box const & root, IReport * out)
  {
  }
},
{
  "A subsample SHALL be created for each tile that has a decode_tile structure whose size (including any trailing bits) is larger or equal to 16 bytes",
  "assert-df6aa49f",
                [](Box const & root, IReport * out)
  {
  }
},
{
  "If it is less than 16 bytes, per the rules above, the decode_tile structure is not encrypted and the corresponding bytes SHOULD be included in the BytesOfClearData field of a surrounding subsample, if any",
  "assert-e145e592",
                [](Box const & root, IReport * out)
  {
  }
},
{
  "All other parts of Tile Group OBUs and Frame OBUs SHALL be unprotected for cenc",
  "assert-f7394188",
                [](Box const & root, IReport * out)
  {
  }
},
{
  "The profile parameter value, represented by a single digit decimal, SHALL equal the value of seq_profile in the Sequence Header OBU",
  "assert-f6db4381",
                [](Box const & root, IReport * out)
  {
  }
},
{
  "The level parameter value SHALL equal the first level value indicated by seq_level_idx in the Sequence Header OBU",
  "assert-d4b4ccd9",
                [](Box const & root, IReport * out)
  {
  }
},
{
  "The tier parameter value SHALL be equal to M when the first seq_tier value in the Sequence Header OBU is equal to 0, and H when it is equal to 1",
  "assert-67559f48",
                [](Box const & root, IReport * out)
  {
  }
},
{
  "The bitDepth parameter value SHALL equal the value of BitDepth variable as defined in [AV1] derived from the Sequence Header OBU",
  "assert-bcdf4133",
                [](Box const & root, IReport * out)
  {
  }
},
{
  "The monochrome parameter value, represented by a single digit decimal, SHALL equal the value of mono_chrome in the Sequence Header OBU",
  "assert-dfda6ebb",
                [](Box const & root, IReport * out)
  {
  }
},
{
  "The chromaSubsampling parameter value, represented by a three-digit decimal, SHALL have its first digit equal to subsampling_x and its second digit equal to subsampling_y",
  "assert-cc705e14",
                [](Box const & root, IReport * out)
  {
  }
},
{
  "If both subsampling_x and subsampling_y are set to 1, then the third digit SHALL be equal to chroma_sample_position, otherwise it SHALL be set to 0",
  "assert-63b6ae0e",
                [](Box const & root, IReport * out)
  {
  }
},
{
  "If a colr box with colour_type set to nclx is present, the colorPrimaries, transferCharacteristics, matrixCoefficients, and videoFullRangeFlag parameter values SHALL equal the values of matching fields in the colr box",
  "assert-b1572a5e",
                [](Box const & root, IReport * out)
  {
  }
},
{
  "The videoFullRangeFlag parameter value SHALL equal the color_range flag in the Sequence Header OBU",
  "assert-a6380b25",
                [](Box const & root, IReport * out)
  {
  }
},
{
  "If the color_description_present_flag is set to 1 in the Sequence Header OBU, the colorPrimaries, transferCharacteristics, and matrixCoefficients parameter values SHALL equal the values of matching fields in the Sequence Header OBU",
  "assert-c158181d",
                [](Box const & root, IReport * out)
  {
  }
},
{
  "The colorPrimaries, transferCharacteristics, and matrixCoefficients parameter values SHOULD be set to the default values below",
  "assert-2644beb9",
                [](Box const & root, IReport * out)
  {
  }
},
{
  "If the codecs parameter string ends with .0.110.01.01.01.0 (containing all the default values below), that trailing part of the string SHOULD be omitted",
  "assert-c1fdc424",
                [](Box const & root, IReport * out)
  {
  }
},
{
  "The parameters sample entry 4CC, profile, level, tier, and bitDepth are all mandatory fields",
  "assert-041ae5cf",
                [](Box const & root, IReport * out)
  {
  }
},
{
  "All the other fields (including their leading '.') are optional, mutually inclusive (all or none) fields",
  "assert-426c6407",
                [](Box const & root, IReport * out)
  {
  }
},
```

**Comment by podborski** [comment]
Is there an easy way to combine this output with the existing implementation?

**Comment by rbouqueau** [comment]
Hi, thank you. Could you resubmit your PR against the master branch? Since av1isobmff ([was merged](https://github.com/gpac/ComplianceWarden/pull/71)), the current configuration makes it difficult to review. Thanks!

**Comment by rbouqueau** [comment]
> Is there an easy way to combine this output with the existing implementation?

Unfortunately no. We should think about a way to update (that would just insert the assert-ids when done for the first time). Any idea?

**Comment by podborski** [comment]
> > Is there an easy way to combine this output with the existing implementation?
> 
> Unfortunately no. We should think about a way to update (that would just insert the assert-ids when done for the first time). Any idea?

Not sure, one way would be to see if we can automatically map somehow. I just did change a couple manually after running: `python aomrules.py --spec av1isobmff -i https://aomediacodec.github.io/av1-isobmff/v1.2.0.html --dump
`

**Comment by podborski** [comment]
Unfortunately I don't have time to finish this mapping myself at the moment. I will be available for any related work next month. If there is anyone who can sit down and map the implemented rules to the existing assert-ids from v1.2.0 this would be cool. If not, I'll try to do it in February.

**Comment by cconcolato** [comment]
FYI, the build process of the spec now generates and publishes the conformance report using ComplianceWarden, see https://aomediacodec.github.io/av1-isobmff/conformance/

File: src/specs/av1_isobmff/av1_isobmff.cpp [hunk 1]
```diff
@@ -465,6 +466,7 @@ const SpecDesc specAv1ISOBMFF = {
       } },
     { "Section 2.1\n"
       "It SHALL contain at least one track using an AV1SampleEntry",
```
Comment (podborski) [inline src/specs/av1_isobmff/av1_isobmff.cpp:468]: "It SHALL contain at least one track using an AV1SampleEntry, possibly transformed by encryption as specified in § 4 Common Encryption",
  "assert-bd1c6212",

File: src/specs/av1_isobmff/av1_isobmff.cpp [hunk 2]
```diff
@@ -510,9 +511,8 @@ const SpecDesc specAv1ISOBMFF = {
         out->covered();
       } },
     { "Section 2.2.4\n"
-      "The width and height in the TrackHeaderBox SHOULD equal, respectively, the maximum\n"
-      "RenderWidth, called MaxRenderWidth, and the maximum RenderHeight, called\n"
-      "MaxRenderHeight, of all the frames associated with this sample entry",
+      "The width and height in the TrackHeaderBox SHOULD equal, respectively, the maximum RenderWidth, called MaxRenderWidth, and the maximum RenderHeight, called MaxRenderHeight, of all the frames associated with this sample entry.",
```
Comment (podborski) [inline src/specs/av1_isobmff/av1_isobmff.cpp:514]: "The width and height in the TrackHeaderBox SHOULD be equal to MaxRenderWidth and MaxRenderHeight, respectively.",
  "assert-b8425a84",

**Comment by podborski** [comment]
We now have AV1-ISOBMFF version 1.3.0 the we should use instead.

**Pull request merged**